src/apps/activity_assignments/api.py

```diff
@@ -31,3 +31,19 @@ async def assignments_create(
             assignments=assignments,
         )
     )
+
+
+async def applet_assignments(
+    applet_id: uuid.UUID,
+    user: User = Depends(get_current_user),
+    session=Depends(get_session),
+) -> Response[ActivitiesAssignments]:
+    await CheckAccessService(session, user.id).check_applet_activity_assignment_access(applet_id)
+    assignments = await ActivityAssignmentService(session).get_all(applet_id)
+
+    return Response(
+        result=ActivitiesAssignments(
+            applet_id=applet_id,
+            assignments=assignments,
```

Comment:
@rcmerlo how much work would it be to support filtering results of this endpoint by arrays of `activityIds` and `activityFlowIds`? I'm just thinking that as MindLogger scales, this endpoint has the potential to return a lot of results if there are a lot of activities & assignments in an applet. When the FE is calling this endpoint, it's likely only interested in assignments associated with a subset of activities and activity flows (only the ones selected in the activities list), so it would be nice to filter out the unrelated ones on the BE.

This is what I'm thinking, as a possible request body that the FE could pass:

```json
{
  "filter": {
    "activityIds": [
      "3fa85f64-5717-4562-b3fc-2c963f66afa6",
      "3fa85f64-5717-4562-b3fc-2c963f66afa6"
    ],
    "activityFlowIds": [
      "3fa85f64-5717-4562-b3fc-2c963f66afa6",
    ]
  }
}
```

It could also pass the filter as comma-separated query parameters, if that makes more sense.

If you think it's too much work, or if you have another opinion, happy to hear it!

Reply:
yeah, we should do that, I will follow what is done in another endpoints using "filtering" classes

Reply:
added as comma-separated query string parameter following other endpoints pattern

Reply:
Thanks for adding the support. Unfortunately, if I provide **both** `activities` and `flows` filters, the BE treats it like an "AND". I need those filters to be treated as "OR" – i.e., if any of the activities match, OR if any of the flows match. Else the FE would have to issue 2 separate requests, which is not ideal.

Maybe the solution would be to just provide a single filter parameter – `activity_or_flow_ids` – rather than both?

Reply:
I fix this issue without change behavior for FE